src/cmp-ui.ts

```diff
@@ -1,25 +1,55 @@
-import { readGuCookie, readIabCookie } from './cookies';
-import { CMP_DOMAIN, CMP_READY_MSG, CMP_CLOSE_MSG } from './config';
+import { readIabCookie } from './cookies';
+import { logConsent } from './consent-logs';
+import {
+    CMP_DOMAIN,
+    CMP_READY_MSG,
+    CMP_CLOSE_MSG,
+    CMP_SAVED_MSG,
+} from './config';
 
-type Callback = () => void;
+type Callback = (error?: Error) => void;
 
 export const setupMessageHandlers = (
     onReadyCmp: Callback,
     onCloseCmp: Callback,
+    onErrorCmp: Callback,
 ): void => {
     const receiveMessage = (event: MessageEvent): void => {
+        const withError = (callback: Callback): void => {
```

Comment:
This might be better named as something like "with errors handled" or even "without errors", given its functionality?